cmd/commands/client/command_client.go

```diff
@@ -42,8 +43,9 @@ func NewCommandWith(
 
 	identityManager := identity.NewIdentityManager(keystoreInstance)
 
-	dialogEstablisherFactory := func(myID identity.Identity) communication.DialogEstablisher {
-		return nats_dialog.NewDialogEstablisher(myID, identity.NewSigner(keystoreInstance, myID))
+	dialogFactory := func(consumerID, providerID identity.Identity, contact dto.Contact) (communication.Dialog, error) {
```

Comment:
This dialogCreator is called from client connection manager, where consumer and provider is always know. Simply do one function call to create dialog instead of two inside connection manager